### PR TITLE
Update no_warning_call utility in tests

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

The version in deprecated_api/ was better than this one. Was facing some false-negative cases where a warning should be raised, but it wasn't raised because most of them are UserWarning and it usually pops out the first one all the time.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: tests/deprecated_api/__init__.py [hunk 1]
```diff
@@ -25,28 +25,6 @@ def _soft_unimport_module(str_module):
         del sys.modules[str_module]
 
 
-@contextmanager
-def no_warning_call(expected_warning: Type[Warning] = UserWarning, match: Optional[str] = None):
```
Comment (carmocca) [inline tests/deprecated_api/__init__.py:29]: Mmm I think we had this accidentally duplicated

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by akihironitta — APPROVED** [review]

File: tests/helpers/utils.py [hunk 2]
```diff
@@ -114,15 +114,22 @@ def inner_f(queue, **kwargs):
...
         except AssertionError:
             # no warning raised
             return
-        raise AssertionError(f"`{warning_type}` was raised: {w}")
+    else:
+        for w in record.list:
+            if w.category is expected_warning and match in w.message.args[0]:
+                break
+        else:
+            return
```
Comment (Borda) [inline tests/helpers/utils.py:132]: this is a bit taff to follow :rabbit: not everyone know meaning `else` for `for` loop

**Review by Borda — APPROVED** [review]

File: tests/helpers/utils.py (hunk 2 above)
Comment (rohitgr7) [reply tests/helpers/utils.py:132]: if they don't, should search and mark it as TIL :smile:

**Pull request merged**